community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/resume.py

```diff
@@ -320,6 +324,13 @@ def start_tpu(data):
 
 def resume_nodes(nodes: List[str], resume_data: Optional[ResumeData]):
     """resume nodes in nodelist"""
+    # Prevent inactive nodes associated with a future reservation from being resumed
+    active_nodes = nodes[:]
+    for i in range(len(nodes)):
+        if (fr := lookup().future_reservation(lookup().node_nodeset(nodes[i]))) and not fr.active_reservation:
+            del active_nodes[i]
+
```

Comment:
We shouldn't just skip them:
* log.warning
* down_nodes_notify_jobs(dormant_fr_nodes, "bla-bla reservation is not here")